software/spmd/bsg_cuda_lite_runtime/memcpy/kernel_memcpy.c

```diff
@@ -0,0 +1,30 @@
+// This kernel performs a memcpy from arrya src to 
+// array dst in DRAM with the given size (in words)
+
+#include "bsg_manycore.h"
+#include "bsg_set_tile_x_y.h"
+
+#define BSG_TILE_GROUP_X_DIM bsg_tiles_X
+#define BSG_TILE_GROUP_Y_DIM bsg_tiles_Y
+#include "bsg_tile_group_barrier.h"
+INIT_TILE_GROUP_BARRIER(r_barrier, c_barrier, 0, bsg_tiles_X-1, 0, bsg_tiles_Y-1);
+
+
+
+int  __attribute__ ((noinline)) kernel_memcpy(int *dst, int *src, int size) {
+
+	int __bsg_tile_group_id = (__bsg_tile_group_id_y * __bsg_grid_dim_x + __bsg_tile_group_id_x);
```

Comment:
Does every invocation have to do this computation? Why not have it computed in the launch function?

Reply:
(This is probably a separate PR, not to be resolve here)

Reply:
Good point! I actually made a PR on bsg_f1 (#425) to add the __bsg_tile_group_id symbol to the binary, and have it set by the host. It went stale at some point, but I'll rebase it and make a PR again, once that is merged in I can go ahead and remove these invocations, (I intentionally named the variable the same as symbol so once that PR is merged in this throws and error so I know to fix it.)